src/plugins/plot/chart/MctChart.vue

```diff
@@ -603,6 +603,9 @@ export default {
             const mainYAxisId = this.config.yAxis.get('id');
             //There has to be at least one yAxis
             const yAxisIds = [mainYAxisId].concat(this.config.additionalYAxes.map(yAxis => yAxis.get('id')));
+            // Save the first drawable y-axis so that we only draw the marquee on that axis
+            this.firstDrawableAxis = yAxisIds.filter(axisId => this.canDraw(axisId))[0];
```

Comment:
This works but the canDraw condition is now checked twice which is unnecessary.
Perhaps something like this?
```
            yAxisIds.filter(this.canDraw).forEach((id, yAxisIndex) => {
                    this.updateViewport(id);
                    this.drawSeries(id);
                    if (yAxisIndex === 0) {
                        this.drawRectangles(id);
                    }
                    this.drawHighlights(id);
                    // only draw these in fixed time mode or plot is paused
                    if (this.annotationViewingAndEditingAllowed) {
                        this.drawAnnotatedPoints(id);
                        this.drawAnnotationSelections(id);
                    }
            });
},
```